src/util/desktophelper.cpp

```diff
@@ -0,0 +1,169 @@
+            return "konqueror --select \"%1\"";
+        } else if (output == "Thunar.desktop" ||
+                output == "Thunar-folder-handler.desktop") {
+            return kSelectInXfce; // Thunar has no --select option
+        } else {
+            // Known file manager with org.freedesktop.FileManager1 interface
+            // * Nautilus
+            // * Nemo
+            // * Deepin File Manager
+            // * Swordfish
+            // * Caja
+            // No solution for
+            // * pcmanfm
+            return kSelectInFreedesktop; // no special command, try Freedesktop and fallback to QDesktopServices";
+        }
+#else
+        return ""; // no special command use QDesktopServices";
+#endif
+    }
+
+    QString removeChildDir(const QString& path) {
+        int index = path.lastIndexOf(QChar('/'));
+        return path.left(index);
+    }
+
+#ifdef Q_OS_LINUX
+    bool selectInFreedesktop(const QString& path) {
+        const QUrl fileurl = QUrl::fromLocalFile(path);
+        const QStringList args(fileurl.toString());
+        QDBusMessage msg = QDBusMessage::createMethodCall(
+                "org.freedesktop.FileManager1",
+                "/org/freedesktop/FileManager1",
+                "org.freedesktop.FileManager1",
+                "ShowItems");
+        msg << args << "";
+        const QDBusMessage response = QDBusConnection::sessionBus().call(msg);
+        const bool success = (response.type() != QDBusMessage::MessageType::ErrorMessage);
+        return success;
+    }
+
+    bool selectInXfce(const QString& path) {
+        const QString folder = QFileInfo(path).dir().absolutePath();
+        const QString filename = QFileInfo(path).fileName();
+
+        QDBusMessage msg = QDBusMessage::createMethodCall(
+                "org.xfce.FileManager",
+                "/org/xfce/FileManager",
+                "org.xfce.FileManager",
+                "DisplayFolderAndSelect");
+        msg << folder << filename << "" << "";
+        qDebug() << "Calling:" << msg;
+        const QDBusMessage response = QDBusConnection::sessionBus().call(msg);
+        const bool success = (response.type() != QDBusMessage::MessageType::ErrorMessage);
+        return success;
+    }
+#endif
+
+} // anonymous namespace
+
+
+namespace mixxx {
+
+void DesktopHelper::openInFileBrowser(const QStringList& paths) {
+    if (sSelectInFileBrowserCommand.isNull()) {
+        sSelectInFileBrowserCommand = getSelectInFileBrowserCommand();
+    }
+    QSet<QString> openedDirs;
+    QString dirPath;
+    for (const auto& path: paths) {
+        dirPath = removeChildDir(path);
+        // First try to select the file in file browser
+
+#ifdef Q_OS_LINUX
+        if (sSelectInFileBrowserCommand == kSelectInFreedesktop &&
+                QFile::exists(path)) {
+            if (!openedDirs.contains(dirPath)) {
+                if (selectInFreedesktop(path)) {
+                    openedDirs.insert(dirPath);
+                    break;
```

Comment:
I'm confused -- doesn't the break ignore the rest of the paths in `paths`?

Reply:
Ups, yes good catch.